### DISCUSSION TITLE
合并搜索结果

### DISCUSSION THREAD

**Discussion by lonyhandsome** [milvus-io/milvus#15778] [Q&A and General discussion]
milvus是否支持合并搜索结果,类似mysql的group by功能

业务场景:
一张图片可能有多个人脸特征 需要根据某一人脸特征去搜索 且搜索结果不能重复

**Comment by yhmo** [comment]
暂时没有这个功能，需要客户端在拿到结果后做。

不知这样能否满足：
该业务场景有多张图片，每张图片可能有多个人脸特征，也就是每张图片产生多条向量。
我们给先给每张图片打个数字序号。然后在milvus中建立一个集合，该集合有3个字段：id, face, pic_num，分别代表向量id，人脸特征，所属图片序号。
把多张图片生成的向量输入milvus，连带图片的序号一起输入。
拿某个人脸特征去搜索，搜索的参数这么写：

```
search_param = {
        "data": search_vectors,
        "anns_field": "face",
        "param": {"metric_type": L2},
        "limit": topk,
        "output_fields": ["pic_num"]}
    results = collection.search(**search_param)
```
这里的"output_fields": ["pic_num"] 用来返回图片的序号。
搜索出来的结果是一组 id和向量距离的键值队，另外还会返回id对应的pic_num，客户端可以根据这里返回的pic_num来归类。
